x/stakeibc/keeper/unbonding_records.go

```diff
@@ -87,7 +95,14 @@ func (k Keeper) GetTotalUnbondAmountAndRecordsIds(ctx sdk.Context, chainId strin
 			epochUnbonding.EpochNumber, hostZoneRecord.Status, hostZoneRecord.NativeTokenAmount))
 
 		// We'll unbond all records that have status UNBONDING_QUEUE and have an amount g.t. zero
-		if hostZoneRecord.Status == recordstypes.HostZoneUnbonding_UNBONDING_QUEUE && hostZoneRecord.NativeTokenAmount.GT(sdkmath.ZeroInt()) {
+		if k.ShouldInitiateHostZoneUnbondingRecord(ctx, hostZoneRecord) {
+			// Dynamically calculate the unbonding amount based on the current redemption rate
+			success := k.UpdateNativeTokensForHostZoneUnbondingRecord(ctx, epochUnbonding.EpochNumber, hostZoneRecord, false)
+			if !success {
+				k.Logger(ctx).Error(utils.LogWithHostZone(chainId, "Failed to update native tokens for host zone unbonding record in epoch %d", epochUnbonding.EpochNumber))
+				continue
+			}
+
 			totalUnbonded = totalUnbonded.Add(hostZoneRecord.NativeTokenAmount)
```

Comment:
are you sure this summation will work? I think the native amount might always be 0 here

Reply:
Good eye. Agreed I don't think this will work. @shellvish @sampocs should we run this PR through the dockernet integration tests to make sure the high level flow works? Would be nice to have that guarantee as we review.

Reply:
we should try and clean this up a bit. It's a bit dangerous to update pointers inside functions and then make assumptions in the calling function

In general, I think we should either:
* Return the struct from the function
* Write to state in the updating function, and read that value back in the calling function 

I'll take a peak at what makes the most sense here